numpy/ma/core.py

```diff
@@ -4582,50 +4606,47 @@ def all(self, axis=None, out=None):
         True
 
         """
+        kwargs = {} if keepdims is np._NoValue else {'keepdims': keepdims}
+
         mask = _check_mask_axis(self._mask, axis)
```

Comment:
You're right that `.flat` is what saves us here.

But I also think this suggests a good change, that I make `_check_mask_axis` take a `keepdims` argument. That way if someone ever removes `.flat` usage in the future there won't be surprising problems. I'll make the change.